src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -186,7 +186,9 @@ internal sealed class BuildServerConnection
                         // the server and we need to fall back to the command line.
                         //
                         // Example: https://github.com/dotnet/roslyn/issues/24124
+#pragma warning disable VSTHRD114 // Avoid returning a null Task (False positive: https://github.com/microsoft/vs-threading/issues/637)
                         return null;
+#pragma warning restore VSTHRD114 // Avoid returning a null Task
```

Comment:
What is the value in running the analyzer on this code base? This code never executes in the context of VS. I don't see why we should be following VS threading rules here.

Reply:
This is a naming pattern rule which was already broadly applied to C# code but not VB code. It just happens to be implemented in a package that contains other analyzers which aren't in use.

Reply:
This isn't a naming pattern violation though, it's a functional rule.

Reply:
We don't _have_ to use this rule, but I believe we would generally agree it is correct and not specific to VS code. In most cases, if someone returns `null` from a `Task`-returning method, they actually meant to return `Task.FromResult<T>(null)`.

Also, this bug will be fixed.

Reply:
> This isn't a naming pattern violation though, it's a functional rule.

I got mixed up thinking this comment was a reply to https://github.com/dotnet/roslyn/pull/45628#discussion_r452447247. Previous reply has been replaced by a new one.